Add feature direction to AD docs (#9137)
opensearch-project/documentation-website: _observing-your-data/ad/index.md

## Step 2: Configure the model

> Add features to your detector. A _feature_ is any field in your index that you want to analyze for anomalies. A detector can discover anomalies across one or more features. 

Reviewer: @kaituo: Is this better?

```suggestion
Add features to your detector. A _feature_ is any field in your index that you want to analyze for anomalies, including aggregations or painless scripts. A detector can discover anomalies across one or more features. 
```

Reviewer 2: Still can be improved. A feature is not field. A feature is an aggregation of a field or a painless script.

> You must choose an aggregation method for each feature: `average()`, `count()`, `sum()`, `min()`, or `max()`. The aggregation method determines what constitutes an anomaly. For example, if you choose `min()`, the detector focuses on finding anomalies based on the minimum values of your feature. If you choose `average()`, the detector finds anomalies based on the average values of your feature.

Reviewer: ```suggestion
You choose an aggregation method for each feature: `average()`, `count()`, `sum()`, `min()`, or `max()`. The aggregation method determines what constitutes an anomaly. For example, if you choose `min()`, the detector focuses on finding anomalies based on the minimum values of your feature. If you choose `average()`, the detector finds anomalies based on the average values of your feature. 

You can also use [custom JSON aggregation queries](#configuring-a-model-based-on-a-json-aggregation-query) as an aggregation method. For more information on creating JSON aggregation queries, see [OpenSearch Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/).

```

### Suppressing anomalies with threshold-based rules

In the **Feature selection** pane, you can suppress anomalies by setting rules that define acceptable differences between the expected and actual values, either as an absolute value or a relative percentage. This helps reduce false anomalies caused by minor fluctuations, allowing you to focus on significant deviations.

For example, to detect substantial changes in log volume while ignoring small variations that are not meaningful. Without customized settings, the system might generate false alerts for minor changes, making it difficult to identify true anomalies. By setting suppression rules, you can ignore minor deviations and focus on real anomalous patterns.

To suppress anomalies for deviations of less than 30% from the expected value, you can set the following rules in the feature selection pane:

> - Ignore anomalies for feature logVolume when the actual value is no more than 30% above the expected value.

Reviewer: ```suggestion
- Ignore anomalies when the actual value is no more than 30% above the expected value.
```

> - Ignore anomalies for feature logVolume when the actual value is no more than 30% below the expected value.

Reviewer: ```suggestion
- Ignore anomalies when the actual value is no more than 30% below the expected value.
```

The following image shows where to set the relative deviation percentage settings:

<img src="{{site.url}}{{site.baseurl}}/images/anomaly-detection/add-feature-with-relative-rules.png" alt="Interface of adding a feature with suppression rules" width="800" height="800">

If you expect that the log volume should differ by at least 10,000 from the expected value before being considered an anomaly, you can set the following absolute thresholds:

> - Ignore anomalies for feature logVolume when the actual value is no more than 10000 above the expected value.

Reviewer: ```suggestion
- Ignore anomalies for feature `logVolume` when the actual value is no more than 10000 above the expected value.
```

> - Ignore anomalies for feature logVolume when the actual value is no more than 10000 below the expected value.

Reviewer: ```suggestion
- Ignore anomalies for feature `logVolume` when the actual value is no more than 10000 below the expected value.
```

For each configured feature, you can also select the anomaly criteria. By default, the model detects an anomaly when the actual value is either abnormally higher or lower than the expected value. However, you can customize your feature settings so that anomalies are only registered when the actual value is higher than the expected value (indicating a spike in the data) or lower than the expected value (indicating a dip in the data). For example, when creating a detector for the `cpu_utilization` field, you may choose to register anomalies only when the value spikes in order to reduce alert fatigue.

### Suppressing anomalies with threshold-based rules

In the **Feature selection** pane, you can suppress anomalies by setting rules that define acceptable differences between the expected and actual values, either as an absolute value or a relative percentage. This helps reduce false anomalies caused by minor fluctuations, allowing you to focus on significant deviations.

> 

Reviewer 3: ```suggestion
```

To suppress anomalies for deviations of less than 30% from the expected value, you can set the following rules in the feature selection pane:

- Ignore anomalies when the actual value is no more than 30% above the expected value.
- Ignore anomalies when the actual value is no more than 30% below the expected value.

> The following image shows where to set the relative deviation percentage settings:

Reviewer: ```suggestion
The following image shows the pane for a feature named `LogVolume`,  where you can set the relative deviation percentage settings:
```

<img src="{{site.url}}{{site.baseurl}}/images/anomaly-detection/add-feature-with-relative-rules.png" alt="Interface of adding a feature with suppression rules" width="800" height="800">

If you expect that the log volume should differ by at least 10,000 from the expected value before being considered an anomaly, you can set the following absolute thresholds:

- Ignore anomalies when the actual value is no more than 10,000 above the expected value.
- Ignore anomalies when the actual value is no more than 10,000 below the expected value.

> The following image shows where to set the absolute threshold settings:

Reviewer: ```suggestion
The following image shows the pane for a feature named `LogVolume`, where you can set the absolute threshold settings:
```

## Step 2: Configure the model

> Add features to your detector. A _feature_  an aggregation of a field or a painless script. A detector can discover anomalies across one or more features. 

Reviewer: ```suggestion
Add features to your detector. A _feature_  an aggregation of a field or a Painless script. A detector can discover anomalies across one or more features. 
```

You choose an aggregation method for each feature: `average()`, `count()`, `sum()`, `min()`, or `max()`. The aggregation method determines what constitutes an anomaly. For example, if you choose `min()`, the detector focuses on finding anomalies based on the minimum values of your feature. If you choose `average()`, the detector finds anomalies based on the average values of your feature. 

> You can also use [custom JSON aggregation queries](#configuring-a-model-based-on-a-json-aggregation-query) as an aggregation method. For more information on creating JSON aggregation queries, see [OpenSearch Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/).

Reviewer: ```suggestion
You can also use [custom JSON aggregation queries](#configuring-a-model-based-on-a-json-aggregation-query) as an aggregation method. For more information about creating JSON aggregation queries, see [OpenSearch Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/).
```

To suppress anomalies for deviations of less than 30% from the expected value, you can set the following rules in the feature selection pane:

- Ignore anomalies when the actual value is no more than 30% above the expected value.
- Ignore anomalies when the actual value is no more than 30% below the expected value.

The following image shows where to set the relative deviation percentage settings:

<img src="{{site.url}}{{site.baseurl}}/images/anomaly-detection/add-feature-with-relative-rules.png" alt="Interface of adding a feature with suppression rules" width="800" height="800">

If you expect that the log volume should differ by at least 10,000 from the expected value before being considered an anomaly, you can set the following absolute thresholds:

- Ignore anomalies when the actual value is no more than 10000 above the expected value.
- Ignore anomalies when the actual value is no more than 10000 below the expected value.

> The following image shows where to set the absolute threshold settings:

Reviewer 3: Name what is shown in the image: "The following image shows the X page/window/pane, where you can set the absolute threshold settings:"

## Step 2: Configure the model

> Add features to your detector. A _feature_ is any field in your index that you want to analyze for anomalies. A detector can discover anomalies across one or more features. 

Reviewer 2: I think current description is not accurate. A feature can be not only an aggregation of a field but also a painless script.

> You must choose an aggregation method for each feature: `average()`, `count()`, `sum()`, `min()`, or `max()`. The aggregation method determines what constitutes an anomaly. For example, if you choose `min()`, the detector focuses on finding anomalies based on the minimum values of your feature. If you choose `average()`, the detector finds anomalies based on the average values of your feature.

Reviewer 2: Can you add if users want to use any other method outside the 5 predefined method like average(), they can use a custom expression (a.k.a., JSON aggregation query below)?

### Suppressing anomalies with threshold-based rules

> In the Feature selection pane, you can suppress anomalies by setting rules that define acceptable differences between the expected and actual values, either as an absolute value or a relative percentage. This helps reduce false anomalies caused by minor fluctuations, allowing you to focus on significant deviations.

Reviewer 2: Please move the suppression rules outside of its own to feature section. that makes it easier to find.

Reviewer 2: you also need to update example snapshot in my example.

Author: I removed this:

"Ensure that a feature, for example, logVolume, is properly defined in your model. Suppression rules are tied to specific features."

There is no photo snapshot currently in the docs but do you mean removing this:
```
Ignore anomalies for feature logVolume when the actual value is no more than 30% above the expected value.
Ignore anomalies for feature logVolume when the actual value is no more than 30% below the expected value.
```

and replace with screenshot of current UI?

Reviewer 2: yeah, i think that would be more straightforward using screenshot.

Author: added 2 screenshots, still kept this wording though. 

First screenshot has entire feature configuration and second just a zoom in of the suppression rules, let me know if you think its too much

Reviewer 2: In images/anomaly-detection/add-suppression-rules-absolute.png, you need use "Units" in both rules.

Author: fixed
